behaviour/graql/reasoner/relation-inference.feature

```diff
@@ -0,0 +1,1299 @@
+      """
+      match
+        $x has name "Haikal";
+        ($x) isa employment;
+      get;
+      """
+    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then for graql query
+      """
+      match
+        $x has name "Michael";
+        ($x) isa employment;
+      get;
+      """
+    Then answer size in reasoned keyspace is: 0
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: a rule can infer a relation with an attribute as a roleplayer
+    Given for each session, graql define
+      """
+      define
+      item sub entity, has name, plays listed-item;
+      price sub attribute, value double, plays item-price;
+      item-listing sub relation, relates listed-item, relates item-price;
+      nutella-price sub rule,
+      when {
+        $x isa item, has name "3kg jar of Nutella";
+        $y 14.99 isa price;
+      }, then {
+        (listed-item: $x, item-price: $y) isa item-listing;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa item, has name "3kg jar of Nutella";
+      $y 14.99 isa price;
+      """
+    When materialised keyspace is completed
+    Then for graql query
+      """
+      match
+        $r (listed-item: $i, item-price: $p) isa item-listing;
+        $i isa item, has name $n;
+        $n "3kg jar of Nutella" isa name;
+        $p 14.99 isa price;
+      get;
+      """
+    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: a rule can infer a relation based on ownership of any instance of a specific attribute type
+    Given for each session, graql define
+      """
+      define
+      year sub attribute, value long, plays favourite-year;
+      employment relates favourite-year;
+      kronenbourg-employs-anyone-with-a-name sub rule,
+      when {
+        $x isa company, has name "Kronenbourg";
+        $p isa person, has name $n;
+        $y 1664 isa year;
+      }, then {
+        (employee: $p, employer: $x, favourite-year: $y) isa employment;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa company, has name "Kronenbourg";
+      $p isa person, has name "Ronald";
+      $p2 isa person, has name "Prasanth";
+      $y 1664 isa year;
+      """
+    When materialised keyspace is completed
+    Then for graql query
+      """
+      match
+        $x 1664 isa year;
+        ($x, employee: $p, employer: $y) isa employment;
+      get;
+      """
+    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 2
+    Then materialised and reasoned keyspaces are the same size
+
+
+  ###############
+  # REFLEXIVITY #
+  ###############
+
+  Scenario: when inferring relations on all pairs from n concepts, the number of relations is the nth triangle number
```

Comment:
comment on what a triangle number is

Reply:
ok keep triangle number

Reply:
Added some comments detailing how the answer is obtained and the definition of a triangle number